src/video/windows/SDL_windowsevents.c

```diff
@@ -1534,15 +1534,6 @@ WIN_WindowProc(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam)
                 suggestedRect->left, suggestedRect->top, suggestedRect->right - suggestedRect->left, suggestedRect->bottom - suggestedRect->top);
 #endif
 
-            if (data->videodata->dpi_scaling_enabled) {
-                /* Update the cached DPI value for this window */
-                data->scaling_dpi = newDPI;
-
-                /* Send a SDL_WINDOWEVENT_SIZE_CHANGED saying that the client size (in dpi-scaled points) is unchanged. 
-                   Renderers need to get this to know that the framebuffer size changed. */
-                SDL_SendWindowEvent(data->window, SDL_WINDOWEVENT_SIZE_CHANGED, data->window->w, data->window->h);
-            }
-
             if (data->expected_resize) {
                 /* This DPI change is coming from an explicit SetWindowPos call within SDL.
                    Assume all call sites are calculating the DPI-aware frame correctly, so
```

Comment:
We'll need to also run this code here, inside the `if (data->expected_resize) { ... }` 

```
            if (data->videodata->dpi_scaling_enabled) {
                /* Update the cached DPI value for this window */
                data->scaling_dpi = newDPI;

                /* Send a SDL_WINDOWEVENT_SIZE_CHANGED saying that the client size (in dpi-scaled points) is unchanged. 
                   Renderers need to get this to know that the framebuffer size changed. */
                SDL_SendWindowEvent(data->window, SDL_WINDOWEVENT_SIZE_CHANGED, data->window->w, data->window->h);
            }
```

Without this, `SDL_WINDOWS_DPI_SCALING=1 testwm2` and using the Alt+Right/Left shortcuts to move the window between monitors will cause unwanted growth/shrinking.

Reply:
Should be good now.